sys/include/net/eui48.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_eui48 IEEE EUI-48
+ * @ingroup     net
+ * @brief       Definition and IPv6 IID conversion for IEEE EUI-48 addresses
+ * @{
+ *
+ * @file
+ * @brief       Definition and IPv6 IID conversion for IEEE EUI-48 addresses
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NET_EUI48_H
+#define NET_EUI48_H
+
+#include <stdint.h>
+
+#include "net/eui64.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Data type to represent an EUI-48
+ */
+typedef union {
+    uint8_t uint8[6];            /**< split into 6 8-bit words. */
+    network_uint16_t uint16[3];  /**< split into 3 16-bit words. */
+} eui48_t;
+
+/**
+ * @brief   Generates an IPv6 interface identifier from a 48-bit device address
+ *
+ * @see <a href="https://tools.ietf.org/html/rfc2464#section-4">
+ *          E.g.: RFC 2464, section 4
+ *      </a>
```

Comment:
BTW, the reasoning for the flipping of the U/L bit can be found in [RFC 4291, section 2.5.1](https://tools.ietf.org/html/rfc4291#section-2.5.1), for those interested.

Reply:
added this also as reference